config/jobs/periodic/docker-in-docker/periodic-dind-build.yaml

```diff
@@ -0,0 +1,213 @@
+periodics:
+  - name: docker-build
+    labels:
+      preset-secrets: "true"
+    cluster: k8s-ppc64le-cluster
+    decorate: true
+    interval: 24h
+    spec:
+      containers:
+      - image: quay.io/florencepascual/docker_ce_build
+        command:
+        - /bin/bash
+        args:
+        - -c
+        - |
+          # path to the github repository
+          DIR_GITHUB="docker-ce-build"
+          PATH_GITHUB="https://github.com/florencepascual/${DIR_GITHUB}.git"
+          
+          # path to the image for building and testing
+          PATH_IMAGE_BUILD="quay.io/florencepascual/docker_ce_build"
+
+          # path to the scripts 
+          PATH_SCRIPTS="/workspace/${DIR_GITHUB}"
+          DATE=`date +%d%m%y-%H%S`
+          PATH_LOG_PROWJOB="/workspace/prow-job-${DATE}.log"
+
+          export DIR_GITHUB
+          export PATH_SCRIPTS
+          export PATH_LOG_PROWJOB
```

Comment:
Have you tried configuring COS details in the job yaml so that the LOG of PROWJOB need not be explicitly written?(Like in https://github.com/ppc64le-cloud/test-infra/blob/master/config/jobs/periodic/kubernetes/test-kubernetes-periodics.yaml#L6)

Checking for various conditions and appending the logs can be altogether avoided if your bucket is configured in the job. When there is an error, the pod would exit on its own.